sdk/tx-kernel/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+#![no_std]
+
+#[panic_handler]
+fn my_panic(_info: &core::panic::PanicInfo) -> ! {
+    loop {}
+}
+
+#[global_allocator]
+static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+// use miden_sdk_types::Felt;
+
+#[repr(transparent)]
+#[derive(Copy, Clone)]
+pub struct AccountId(Felt);
+
+impl From<AccountId> for Felt {
+    fn from(account_id: AccountId) -> Felt {
+        account_id.0
+    }
+}
+
+#[link(wasm_import_module = "miden:tx_kernel/account")]
+extern "C" {
+    #[link_name = "get_id<0x0000000000000000000000000000000000000000000000000000000000000000>"]
+    fn extern_account_get_id() -> AccountId;
+    #[link_name = "add_asset<0x0000000000000000000000000000000000000000000000000000000000000000>"]
+    fn extern_account_add_asset(_: Felt, _: Felt, _: Felt, _: Felt, _: i32);
+}
+
+#[link(wasm_import_module = "miden:tx_kernel/note")]
+extern "C" {
+    #[link_name = "get_inputs<0x0000000000000000000000000000000000000000000000000000000000000000>"]
+    fn extern_note_get_inputs(ptr: i32) -> i32;
```

Comment:
The `ptr` value here can be `*mut u8`. It will have the correct type when lowered to Wasm (i.e. the Wasm `i32` type). We should always try to use the most accurate type possible that will have the correct C ABI representation to match what is expected by the callee (rather than trying to represent how we translate the C ABI types to Miden types here).

Reply:
Also, this API can return `usize`, since we know that `usize` is equivalent to `u32` for Miden. That simplifies some of the other code that works with this API, i.e. no redundant casts.

Reply:
Thanks! Done in https://github.com/0xPolygonMiden/compiler/pull/140/commits/7b21de4a54a1f79834946685a9cf8a95254473c0